ddl/ddl_api.go

```diff
@@ -273,6 +274,14 @@ func checkColumnDefaultValue(ctx sessionctx.Context, col *table.Column, value in
 		// In strict SQL mode or default value is not an empty string.
 		return hasDefaultValue, value, errBlobCantHaveDefault.GenWithStackByArgs(col.Name.O)
 	}
+	if value != nil && ctx.GetSessionVars().SQLMode.HasNoZeroDateMode() && ctx.GetSessionVars().SQLMode.HasStrictMode() && (col.Tp == mysql.TypeTimestamp ||
+		col.Tp == mysql.TypeDatetime || col.Tp == mysql.TypeDate) {
+		if vv, ok := value.(string); ok {
+			if vv == types.ZeroDatetimeStr || vv == types.ZeroDateStr {
```

Comment:
What if below sql:
```sql
create table t1 (a timestamp default '0000-00-00 00');
```

Reply:
This sql is not allow by mysql, but pass this pr. i will fix it.

Reply:
Please add unit test for this.

Reply:
@csuzhangxc addressed.